### PR TITLE
[REVIEW] feature/robust_service_start

### PR DESCRIPTION
**Pull request by williamBlazing**
PLEASE CHECK:
Please check how i am shutting down the calcite, i left it really basic, but we could improve that

### CODE REVIEW

**Comment by kharoc** [comment]
**READY TO REVIEW!**
Single Node: OK
Distributed Mode: OK

File: blazingdb-calcite-application/src/main/java/com/blazingdb/calcite/application/CalciteService.java [hunk 1]
```diff
@@ -84,6 +86,31 @@ public static ByteBuffer processRequest(ByteBuffer buffer, final String dataDire
 
 			}
 			return response.getBufferData();
+		
+		} else if (requestMessage.getHeaderType() == MessageType.SystemCommand) {
+			
+			ResponseMessage response = null;
+
+			SystemCommandRequestMessage message = new SystemCommandRequestMessage(requestMessage.getPayloadBuffer());
+			if (message.getCommand().contains("shutdown")){
```
Comment (felipeblazing) [inline blazingdb-calcite-application/src/main/java/com/blazingdb/calcite/application/CalciteService.java]: remove string parsing and use an enum to handle the different commands that can be sent

File: blazingdb-calcite-application/src/main/java/com/blazingdb/calcite/application/TCPService.java [hunk 2]
```diff
@@ -57,30 +57,46 @@ public TCPService(final Integer tcpPort, final String dataDirectory, String orch
...
-			if (responsePayload.getSchemas().size() > 0) {
-				String dbName = responsePayload.getSchemas().get(0).getDbName();
-				System.out.println("dbName from orch: " + dbName);
-				for (DDLCreateTableRequestMessage schema : responsePayload.getSchemas()) {
-					ApplicationContext.getCatalogService(dataDirectory).createTable(schema);
-					// I am unsure at this point if we have to update the schema or not but for safety I do it here
-					// need to see what hibernate moves around :)
-					ApplicationContext.updateContext(dataDirectory);
+		int num_tries=0;
+		int max_tries=20;
```
Comment (felipeblazing) [inline blazingdb-calcite-application/src/main/java/com/blazingdb/calcite/application/TCPService.java]: why did you pick 20 for calcite and 60 for ral in turms of max_tries?

File: blazingdb-calcite-application/src/main/java/com/blazingdb/calcite/application/TCPService.java [hunk 3]
```diff
@@ -57,30 +57,46 @@ public TCPService(final Integer tcpPort, final String dataDirectory, String orch
...
-
-			if (responsePayload.getSchemas().size() > 0) {
-				String dbName = responsePayload.getSchemas().get(0).getDbName();
-				System.out.println("dbName from orch: " + dbName);
-				for (DDLCreateTableRequestMessage schema : responsePayload.getSchemas()) {
-					ApplicationContext.getCatalogService(dataDirectory).createTable(schema);
-					// I am unsure at this point if we have to update the schema or not but for safety I do it here
-					// need to see what hibernate moves around :)
-					ApplicationContext.updateContext(dataDirectory);
+		int num_tries=0;
```
Comment (felipeblazing) [inline blazingdb-calcite-application/src/main/java/com/blazingdb/calcite/application/TCPService.java]: use camel case in java

File: blazingdb-calcite-application/src/main/java/com/blazingdb/calcite/application/TCPService.java [hunk 4]
```diff
@@ -57,30 +57,46 @@ public TCPService(final Integer tcpPort, final String dataDirectory, String orch
...
+		int max_tries=20;
+		boolean connection_success = false;
```
Comment (felipeblazing) [inline blazingdb-calcite-application/src/main/java/com/blazingdb/calcite/application/TCPService.java]: use camel case

File: blazingdb-calcite-application/src/main/java/com/blazingdb/calcite/application/TCPService.java [hunk 5]
```diff
@@ -122,33 +138,23 @@ public void run() {
...
-
-						
-				    	connectionSocket.getOutputStream().write(i);
-			        }
-				*/
-				      
-				      
-					// outToClient.flush();
+					int value = resultBytes.length;
+					ByteBuffer mybuffer = ByteBuffer.allocate(4);
```
Comment (felipeblazing) [inline blazingdb-calcite-application/src/main/java/com/blazingdb/calcite/application/TCPService.java]: dont use the name mybuffer it doesnt tell us much

File: blazingdb-calcite-application/src/main/java/com/blazingdb/calcite/application/TCPService.java [hunk 6]
```diff
@@ -122,33 +138,23 @@ public void run() {
...
+					mybuffer.order(ByteOrder.LITTLE_ENDIAN);
+					mybuffer.putInt(value);
+					mybuffer.rewind();
+					byte[] mymyresultBytes = mybuffer.array();
```
Comment (felipeblazing) [inline blazingdb-calcite-application/src/main/java/com/blazingdb/calcite/application/TCPService.java]: definitely dont use mymyresult bytes, if you are changing this code fix the names

**Pull request merged**